Aptoide/aptoide-client-v8
# [APP-2727] - Implement Editorials Logic

**What does this PR do?**

It implements the whole logic of Editorials in home. It also refactors some urls and edits already made views in order for them to match the design in figma.

**Database changed?**

No

**Where should the reviewer start?**

- [ ] AppViewScreen.kt
- [ ] RepositoryModule.kt
- [ ] EditorialMoreView.kt
- [ ] EditorialView.kt
- [ ] EditorialViewScreen.kt
- [ ] WidgetTypeJSON.kt
- [ ] BundlesUseCase.kt
- [ ] Widget.kt
- [ ] ArticlesMetaUseCase.kt
- [ ] Article.kt
- [ ] ArticleListUiState.kt
- [ ] EditorialsListViewModel.kt
- [ ] ViewModelProvider.kt

**How should this be manually tested?**

  Test to see if all the editorial urls work without the widget in home and if they match the design. 
  Test to see if all editorial views (including in home) work with the widget in home. While we don't have the widget in home, you can use the charles rewrite left in the ticket to test.
  
  Flow on how to test this or QA Tickets related to this use-case: [AND-49](https://aptoide.atlassian.net/browse/AND-49)

**What are the relevant tickets?**

  Tickets related to this pull-request: [AND-49](https://aptoide.atlassian.net/browse/AND-49)




**Code Review Checklist**


[AND-49]: https://aptoide.atlassian.net/browse/AND-49?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
